adagrams/game.py

```diff
@@ -1,11 +1,144 @@
+from multiprocessing import pool
+import random
+
+def build_letter_pool():
+    LETTER_POOL = {
+            "A": 9,
+            "B": 2,
+            "C": 2,
+            "D": 4,
+            "E": 12,
+            "F": 2,
+            "G": 3,
+            "H": 2,
+            "I": 9,
+            "J": 1,
+            "K": 1,
+            "L": 4,
+            "M": 2,
+            "N": 6,
+            "O": 8,
+            "P": 2,
+            "Q": 1,
+            "R": 6,
+            "S": 4,
+            "T": 6,
+            "U": 4,
+            "V": 2,
+            "W": 2,
+            "X": 1,
+            "Y": 2,
+            "Z": 1
+            }
+
+    return LETTER_POOL
+
+
+def build_available_letters_list(LETTER_POOL):
+    available_letters = []
+
+    for letter,frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            available_letters.append(letter)
+
+    return available_letters
+
+
 def draw_letters():
-    pass
+    hand = []
+    LETTER_POOL = build_letter_pool()
+    available_letters = build_available_letters_list(LETTER_POOL)
+    count_dict = {}
+    
+    while len(hand) < 10:
+        random_letter = random.choice(available_letters)
+        
+        if random_letter not in hand:
+            hand.append(random_letter)
+            count_dict[random_letter] = 1
+        elif LETTER_POOL[random_letter] > count_dict[random_letter]:
+            hand.append(random_letter)
+            count_dict[random_letter] += 1
+
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = letter_bank.copy()
+    for letter in word.upper():
+        if letter not in letter_bank_copy:
```

Comment:
Let's have a closer look at the time complexity involved with lines 68-69. 

Line 68: a for loop that loops as many times as there are letters in word - O(n)
Line 69: in operator that in the worst case scenario has to check every letter in letter_bank_copy to see if letter is in it - O(n)

Rather than using a list of the letters in the hand, could we build a helper data structure (like a dictionary) that could let us look up if a letter was part of a user's hand?

If the helper dictionary had keys that were the letters and the values were the number of letters in the hand, then you could leverage the fact that key look up in dictionaries is a constant O(1) operation. 

So on line 69, you can check if key (the letter in the hand) in helper_dictionary, which has constant look up time. 

Additionally, instead of using the remove method, you could decrement the value in the dictionary to keep track of how many letters have been played. Recall that the remove method on a list also has linear run time because it could have to iterate over the entire list before finally removing letter.